_posts/2020-09-16-chross-browser-mobile-webapp.markdown

```diff
@@ -0,0 +1,127 @@
+
+## The Notch
+
+iOS introduced `The Notch` a few years ago. This hardware element will cover a part of the screen. Your mini will be displayed behind the notch, so you'll have to take that into consideration when designing your app, so you don't put elements fixed at the top that might be obscured by the notch.
+
+You can use CSS' safe area inset values to add extra padding when needed.
+
+If you need to know these values in your JavaScript code, the SDK provides that information (using `sc.app.safeAreaInsets`) with the top and bottom safe area inset values.
+
+> This is another great article from css-tricks you can use for more in-depth code an examples: [link](https://css-tricks.com/the-notch-and-css/).
+
+## iOS Overscroll
+
+Apple users are used to the extra scroll behavior. When the user reaches the end of something that's scrollable, they can scroll a bit more and a white space shows up in the opposite direction they over-scrolled.
+
+If you want to prevent that overscroll to shift your whole app outside the viewport, you can position your app fixed to the viewport, so if the user overscrolls it, it will scroll to the static element in the background but your app will be fixed on top so the scroll will not affect it. It may be hard to understand, and the code for this is just a `position: fixed` on an element, hopefully this explanation is enough if you are experiencing the same issue.
+
+## Android Keyboard Covers Screen
+
+Android and iOS devices do different things when you open the keyboard. Each of them has pros and cons.
+
+One key difference is that, when you open the keyboard on iOS, it will try to scroll the page so the input is inside the viewport and if it can scroll to it it will move the whole viewport outside the screen if necessary so the input is visible.
+
+For Android, it will only try to scroll but if the element is to look inside the viewport, it will be covered by the keyboard. You may need to take this into consideration when designing and the SDK provides an event to detect when the keyboard was opened or closed so you can react to that and make sure your input is visible.
+
+## Safari can't do Smooth Scroll
+
+Safari doesn't support the value `smooth` for the [scroll-behavior](https://developer.mozilla.org/en-US/docs/Web/CSS/scroll-behavior) property. It also doesn't support the `behavior: 'smooth'` option for the `Window.scrollTo` function.
+
+If you need smooth scrolling you may use [a polyfill](https://www.npmjs.com/package/smoothscroll-polyfill) or implement another custom scroll mechanism.
+
+## Conclusion
+
+Making sure your app behaves similarly on both platforms is important to provide a good user experience. Sometimes it might be hard to reproduce the issues and to test the app in enough devices to debug them, so we hope some of these tips can help you fix them more easily.
```

Comment:
```suggestion
Making sure your app behaves similarly on both platforms is important to provide a good user experience. Sometimes it might be hard to reproduce the issues and to test the app on enough devices to debug them, so we hope some of these tips can help you fix them more easily.
```